controller/lib/core/sensors.cpp

```diff
@@ -71,7 +78,37 @@ static constexpr Duration VOLUME_INTEGRAL_INTERVAL = milliseconds(5);
   __builtin_unreachable();
 }
 
-Sensors::Sensors() {}
+// Tuning params for PID that calculates error in our flow measurements.
+// Calculated using the
+// https://en.wikipedia.org/wiki/Ziegler%E2%80%93Nichols_method The constants in
+// the constructor are for the "classic PID" control type, which seemed to work
+// well.
+//
+// Note that we have to choose a nominal sample period for the PID.  In reality
+// we sample the PID once per breath; the exact value we use for the sample
+// period is not important so long as it's shorter than our breath period.  If
+// we were to choose a value longer than our breath period, then the PID
+// wouldn't update on some breaths, which is not desirable!
+//
+// FLOW_PID_OUTPUT_MAX is a sanity check, preventing us from using absurdly
+// high flow corrections.  100ml/s may strike you as absurdly high, but we have
+// seen justified need to go as high as 40ml/s.
+static constexpr float FLOW_PID_KU = 0.20f;
+static constexpr float FLOW_PID_TU = 5;
+static constexpr Duration FLOW_PID_SAMPLE_PERIOD = seconds(1);
+static constexpr VolumetricFlow FLOW_PID_OUTPUT_MAX = ml_per_sec(100);
+
+Sensors::Sensors()
+    : flow_error_pid_(
+          /*kp=*/0.6f * FLOW_PID_KU,
+          /*ki=*/1.2f * FLOW_PID_KU / FLOW_PID_TU,
+          /*kd=*/3 * FLOW_PID_KU * FLOW_PID_TU / 40,
+          ProportionalTerm::ON_ERROR,        //
```

Comment:
Please document the reasons for choosing ON_ERROR and ON_MEASUREMENT here, as well as JITTER_ALLOWED